src/CreatePostWindow.tsx

```diff
@@ -0,0 +1,90 @@
+import "./CreatePostWindow.css";
+import React, { useState, Component } from "react";
+
+function CreatePostWindow(props: {
+  OnCreate: (_: string) => void;
+  ToggleWindow: () => void;
+  SetTemp: (_: string) => void;
+  isInProcess: (_: boolean) => void;
+  Temp: string;
+  Username: string;
+}) {
+  //   const [input, setInput] = useState("");
+  // if (input!==''||props.TempNew!==''){
+  //     props.isInProcess(true)
+  // }
+
+  const handleSubmit = (e: React.MouseEvent<HTMLButtonElement>) => {
+    e.preventDefault();
+    props.OnCreate(props.Temp);
+    // setInput("");
+    document
+      .getElementsByClassName("opacitySolid")[0]
+      .setAttribute("style", "display:none");
+    // document.getElementsByClassName("create_post_container")[0].setAttribute('style', 'display:none')
+    document.body.style.overflow = "visible";
+    props.SetTemp("");
+    props.isInProcess(false);
+    props.ToggleWindow();
+  };
+
+  //   if (props.Temp.replace(/\s/g, "") !== ""){
+  //     document.getElementsByClassName("create_post__button")[0].setAttribute('style', 'background-color:#2374E1; color:white; cursor:pointer' );
+  //   }
+  const button_styles = {
+    backgroundColor: `${props.Temp.replace(/\s/g, "") !== "" ? "#2374E1" : "#505151" }`,
```

Comment:
i don't understand what do you want with this regex

Reply:
it detects inputs wholly comprised of white spaces, e.g.:
"                     "  ==>  ""

specifically replaces all white spaces with an empty string and then check if equals

Reply:
Ok, I see you do it in some places, try to extract it to a function with a good name and use this function wherever you need

Reply:
I'm using it over a few components. Should I use global function, or pass it as a props?